examples/quantum_volume.py

```diff
@@ -0,0 +1,153 @@
+    # need to classically simulate it to compute its heavy set.
+    return circuit
+
+
+def compute_heavy_set(circuit: cirq.Circuit) -> List[str]:
+    """Classically compute the heavy set of the given circuit.
+
+    The heavy set is defined as the output bit-strings that have a greater than
+    median probability of being generated.
+
+    Args:
+        circuit: The circuit to classically simulate.
+
+    Returns:
+        A list containing all of the heavy bit-string results.
+    """
+    # Classically compute the probabilities of each output bit-string through
+    # simulation.
+    simulator = cirq.Simulator()
+    results = cast(cirq.WaveFunctionTrialResult,
+                   simulator.simulate(program=circuit))
+    # Compute the median probability of the output bit-strings.
+    median = statistics.median(results.state_vector())
+
+    # The output wave function is a vector from the result value (big-endian) to
+    # the probability of that bit-string. Compute a format string converts the
+    # given index to the corresponding qubit string.
+    format_str = '{0:0%sb}' % len(circuit.all_qubits())
+    # Return all of the bit-strings that have a probability greater than the
+    # median.
+    return ([
+        format_str.format(idx)
+        for idx, prob in enumerate(results.state_vector())
+        if prob > median
+    ])
+
+
+def main(num_qubits: int, depth: int, num_repetitions: int):
+    """Run the quantum volume algorithm.
+
+    The Quantum Volume benchmark is fairly straightforward. This algorithm will
+    follow the same format as Algorithm 1 in
+    https://arxiv.org/abs/1811.12926. To summarize, we generate a random model
+    circuit, compute its heavy set, then transpile an implementation onto our
+    architecture. This implementation is run a series of times and if the
+    percentage of outputs that are in the heavy set is greater than 2/3, we
+    consider the quantum volume test passed for that size.
+
+    Args:
+        num_qubits: The number of qubits for the circuit.
+        depth: The number of gate layers to generate.
+        num_repetitions: The number of times to run the algorithm.
+    """
+    for _ in range(num_repetitions):
+        model_circuit = generate_model_circuit(num_qubits, depth)
+        print(model_circuit)
+        heavy_set = compute_heavy_set(model_circuit)
+        print(heavy_set)
+        # TODO(villela): Implement model circuit and run it.
+
+
+def parse_arguments(args):
+    """Helper function that parses the given arguments."""
+    parser = argparse.ArgumentParser('Quantum volume benchmark.')
+    parser.add_argument('--num_qubits',
+                        default=4,
+                        type=int,
+                        help='The number of circuit qubits to benchmark.')
+    parser.add_argument('--depth',
+                        default=4,
+                        type=int,
+                        help='SU(4) circuit depth.')
+    parser.add_argument(
```

Comment:
If you are going to have a seed, take it as an argument.

Reply:
Done.